### DISCUSSION TITLE
Configure Filament to use Auth instead of default admin/login

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2493] [Help] [answered]
Hello,

I'm rather new to Laravel but have been fumbling around a little and have something setup to use [LdapRecord](https://github.com/DirectoryTree/LdapRecord) with the default Auth framework. 

I would [also ](https://github.com/laravel-filament/filament/discussions/1692) would like to configure Filament to use the default Auth scaffolding that I have re-configured to user the Ldap username instead of the e-mail address but the documentation is not very clear on what or how to do that.

 I am just guessing that I should replace the 'login' Class in filament.php to something else?  What would I change it to?
`'login' => \Filament\Http\Livewire\Auth\Login::class,`

Thank you.

**Comment by danharrin** [comment]
the Login class is a Livewire component page, so you can extend it, make the changes in the new class, and then add the new class name in config.

I've seen quite a few people switch to username, so it probably isn't too tricky?

**Reply by yaplej** [reply]
Let me know if I am in the ballpark.  I seem to be a little lost with this.

Looking at /Vendor/Filament/Filament/Src/Http/Livewire/Auth/Login.php this is what the filament.php config file appears to point to.

I should be able to create my own /App/Livewire/Auth/Login.php extend the functionality and overwrite the default login page?  I could simply make it view the Auth /login view instead of default `return view('filament::login')` ?

You mentioned its a components page but I am not seeing any component other than "login.blade.php" and that does not implement any class to extend because its just a blade.

**Reply by danharrin** [reply]
If you already have a login page to use instead, set the config option to `null`, then create a redirect route to your login page named `filament.auth.login`

If you do still want to override and use Filament's login page, sounds like you're doing it right

**Comment by yaplej** [comment]
I was able to display the build-in auth.login view without making any other changes.

Change routes/web.php
```
Route::get('/admin/login', function () {
    return view('auth.login');
})->name('filament.auth.login');
```
or
```
Route::get('/admin/login', function () {
    return redirect('/login');
})->name('filament.auth.login');
```

I will keep trying to modify the build-in Filament login page too just so I can do both.

Thank you.

**Comment by yaplej** [comment]
Hello again,

I have been able to figure out how to customize the login page to display `Username` instead of `Email address`.

I started with `\App\Filament\Pages\Auth\Login.php` from the Demo page but copied the classes and function from `vendor\filament\filament\src\Http\Livewire\Auth\Login.php`.

That part seems to work just fine.

```
<?php

namespace App\Filament\Pages\Auth;

use DanHarrin\LivewireRateLimiting\Exceptions\TooManyRequestsException;
use DanHarrin\LivewireRateLimiting\WithRateLimiting;
use Filament\Facades\Filament;
use Filament\Forms\ComponentContainer;
use Filament\Forms\Components\Checkbox;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Http\Responses\Auth\Contracts\LoginResponse;
use Illuminate\Contracts\View\View;
use Livewire\Component;
use Filament\Http\Livewire\Auth\Login as BasePage;

class Login extends BasePage implements HasForms
{

    public $username = '';
    public $password = '';
    public $remember = false;

    public function authenticate(): ?LoginResponse
    {
        try {
            $this->rateLimit(5);
        } catch (TooManyRequestsException $exception) {
            $this->addError('username', __('filament::login.messages.throttled', [
                'seconds' => $exception->secondsUntilAvailable,
                'minutes' => ceil($exception->secondsUntilAvailable / 60),
            ]));

            return null;
        }

        $data = $this->form->getState();

        if (! Filament::auth()->attempt([
            'username' => $data['username'],
            'password' => $data['password'],
        ], $data['remember'])) {
            $this->addError('username', __('filament::login.messages.failed'));

            return null;
        }

        return app(LoginResponse::class);
    }

    protected function getFormSchema(): array
    {
        return [
            TextInput::make('username')
                ->label(__('Username'))
                ->required()
                ->autocomplete(),
            TextInput::make('password')
                ->label(__('filament::login.fields.password.label'))
                ->password()
                ->required(),
            Checkbox::make('remember')
                ->label(__('filament::login.fields.remember.label')),
        ];
    }

}
```

It seems like the call to `Filament::auth()->attempt` does not work.  No users can actually login with their username instead of their e-mail address.  It keeps returning `These credentials do not match our records.`.  In my laravel.log I can see that Auth Middleware binds to LDAP but the user does not seem to actually try to authenticate.

Thanks for all the help.

**Reply by danharrin** [reply]
Have you set the Filament auth guard to be your LDAP guard? Otherwise, it will not be able to find your user data

**Reply by yaplej** [reply]
I modified the default provider in `auth.php`.  I need to modify something in `filament.php` too?

```
    'providers' => [
        'users' => [
            'driver' => 'ldap',
            'model' => LdapRecord\Models\ActiveDirectory\User::class,
            'rules' => [],
            'database' => [
                'model' => App\Models\User::class,
                'sync_passwords' => false,
                'sync_attributes' => [
                    'name' => 'cn',
                    'username' => 'samaccountname',
                    'email' => 'mail',
                ],
                'sync_existing' => [
                    'name' => 'cn',
                    'username' => 'samaccountname',
                    'email' => 'mail',
                ],
            ],
        ],
```

The Guard is unchanged.
```
    'guards' => [
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],
    ],
```

Just guessing here but if the model being used in my provider is `LdapRecord\Models\ActiveDirectory\User::class` it probably would not implement `FilamentUser` from `use Filament\Models\Contracts\FilamentUser;` or have the `public function canAccessFilament(): bool` function implemented.

**Comment by yaplej** [comment, marked as answer]
I was finally able to figure it out.  I just needed to pass the username as samaccountname instead of username.

```
<?php

namespace App\Filament\Pages\Auth;

use DanHarrin\LivewireRateLimiting\Exceptions\TooManyRequestsException;
use DanHarrin\LivewireRateLimiting\WithRateLimiting;
use Filament\Facades\Filament;
use Filament\Forms\ComponentContainer;
use Filament\Forms\Components\Checkbox;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Http\Responses\Auth\Contracts\LoginResponse;
use Illuminate\Contracts\View\View;
use Livewire\Component;
use Filament\Http\Livewire\Auth\Login as BasePage;

class Login extends BasePage implements HasForms
{

    public $username = '';
    public $password = '';
    public $remember = false;

    public function authenticate(): ?LoginResponse
    {
        try {
            $this->rateLimit(5);
        } catch (TooManyRequestsException $exception) {
            $this->addError('username', __('filament::login.messages.throttled', [
                'seconds' => $exception->secondsUntilAvailable,
                'minutes' => ceil($exception->secondsUntilAvailable / 60),
            ]));

            return null;
        }

        $data = $this->form->getState();

        if (! Filament::auth()->attempt([
            'samaccountname' => $data['username'],
            'password' => $data['password'],
        ], $data['remember'])) {
            $this->addError('username', __('filament::login.messages.failed'));

            return null;
        }

        return app(LoginResponse::class);
    }

    protected function getFormSchema(): array
    {
        return [
            TextInput::make('username')
                ->label(__('Username'))
                ->required()
                ->autocomplete(),
            TextInput::make('password')
                ->label(__('filament::login.fields.password.label'))
                ->password()
                ->required(),
            Checkbox::make('remember')
                ->label(__('filament::login.fields.remember.label')),
        ];
    }
}
```

**Reply by Timi-Php** [reply]
Where can i locate this file itself

**Reply by barbaracrlp** [reply]
Hi, I'm encountering the same problem I want Filament to use my users table , would you please explain to me all the changes you have made? Thanks in Advance

**Reply by jtomek** [reply]
Extend the Login class from Filament's Panel and check how to specify your own view and use login hooks (AdminPanelProvider)

**Comment by Timi-Php** [comment]
Instead of using "Finament::auth()->attempt" Use your own custom method for
authenticating
